sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosTable.scala

```diff
@@ -2,78 +2,73 @@
 // Licensed under the MIT License.
 package com.azure.cosmos.spark
 
-import java.util
-import java.util.UUID
-
-import com.azure.cosmos.implementation.{CosmosClientMetadataCachesSnapshot, RxDocumentClientImpl}
+import com.azure.cosmos.implementation.CosmosClientMetadataCachesSnapshot
 import com.azure.cosmos.models.PartitionKey
-import com.azure.cosmos.{CosmosBridgeInternal, CosmosClientBuilder, CosmosException}
+import com.azure.cosmos.spark.CosmosTableSchemaInferer.{
+  ID_ATTRIBUTE_NAME,
+  RAW_JSON_BODY_ATTRIBUTE_NAME,
+  TIMESTAMP_ATTRIBUTE_NAME
+}
+import com.azure.cosmos.{CosmosAsyncClient, CosmosClientBuilder, CosmosException}
 import org.apache.spark.broadcast.Broadcast
 import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.connector.catalog.{SupportsRead, SupportsWrite, Table, TableCapability}
 import org.apache.spark.sql.connector.expressions.Transform
 import org.apache.spark.sql.connector.read.ScanBuilder
 import org.apache.spark.sql.connector.write.{LogicalWriteInfo, WriteBuilder}
-import org.apache.spark.sql.types.{IntegerType, StringType, StructField, StructType}
+import org.apache.spark.sql.types.{LongType, StringType, StructField, StructType}
 import org.apache.spark.sql.util.CaseInsensitiveStringMap
 import org.codehaus.jackson.node.ObjectNode
-import org.json4s.scalap.scalasig.ClassFileParser.state
-import reactor.core.publisher.Mono
+
+import java.util
+import java.util.UUID
 
 // scalastyle:off underscore.import
 import scala.collection.JavaConverters._
 // scalastyle:on underscore.import
 
+private object CosmosTable {
+  // TODO fabianm - ??? Should we also add a default "_pk" column, which would retrieve the
```

Comment:
I don't believe so, I thought about it, but if the PK is nested, it makes it much harder.

Reply:
Agreed - will remove the TODO - came to the same conclusion later but forgot to remove the comment.